ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -869,6 +880,30 @@ function run_upgrade_e2e() {
     echo "e2e test take $(( ($e2e_end_time - $e2e_start_time)/60 )) minutes"
 }
 
+function set_channel(){
+    local x_ver y_ver version="$1"
+    x_ver=$( echo "${version}" | cut -f1 -d. )
+    y_ver=$( echo "${version}" | cut -f2 -d. )
+    ver="${x_ver}.${y_ver}"
+    target_channel="${UPGRADE_CHANNEL}-${ver}"
+    if ! oc adm upgrade channel ${target_channel}; then
+        echo "Fail to change channel to ${target_channel}!"
+        exit 1
+    fi
+    local retry=3
+    while (( retry > 0 )); do
```

Comment:
I guess we also need to add the while loop to `cucushift-upgrade-setchannel` ?

Reply:
That move-out is because on that time, it is checking image available or not, if no any available image, directly exit as failure, that is I suggest to move it out. But in later Jian Li's testing, introduced this loop is waiting for cvo can get available update for upstream server because of some delay, if no any available update, this would not cause any failure, but just a warning, the purpose of this loop is ensure cvo get connected with Cincinnati server, they are different.

Reply:
so you want it always wait for conditionupdate after set_channel, right? but there is some test does not care about conditionupdate, it will also wait in this scenario. 
personally +1 for your original thought on `set channel is only doing set channel`, and leaving wait or not and waiting for what to the following steps depending on the specified test scenario. 
considering 3min sounds not too bad, so leave it to you and jian

Reply:
Always waiting less than 3 mins should be harmless, so I think it is acceptable.